src/components/credit-card-number/code/views/creditCardNumberView.jsx

```diff
@@ -0,0 +1,78 @@
+var React = require('react');
+var Payment = require('payment');
+
+module.exports = React.createClass({
+
+  intent: null,
+  formatting: false,
+
+  propTypes: {
+    accepted: React.PropTypes.array,
+    cardNumber: React.PropTypes.string,
+    cardType: React.PropTypes.oneOf(['amex', 'discover', 'mastercard', 'visa']),
+    defaultValue: React.PropTypes.string,
+    id: React.PropTypes.string,
+    label: React.PropTypes.string,
+    name: React.PropTypes.string,
+    errorMessage: React.PropTypes.string
+  },
+
+  getInitialState: function() {
+
+    var is_valid = Payment.fns.validateCardNumber(this.props.defaultValue);
+    var card_type = Payment.fns.cardType(this.props.defaultValue);
+
+    return {
+      valid: is_valid,
+      error: (is_valid) ? null : this.props.errorMessage,
+      cardType: card_type
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      accepted: ['amex', 'discover', 'mastercard', 'visa'],
+      formatInput: false,
+      id: 'credit-card-number',
+      name: 'credit-card-number',
+      placeHolder: '• • • •   • • • •   • • • •   • • • •',
+      showImages: false,
+      errorMessage: 'Invalid Card Number'
+    };
+  },
+
+  handleChange: function(elm) {
```

Comment:
Being pedantic here but the argument ```elm``` here makes me thing you are passing an element into the handle change method when in fact it's an event (convention is that it's ```e```)